core/src/main/java/com/capitalone/dashboard/model/Component.java

```diff
@@ -42,8 +42,28 @@ public void setOwner(String owner) {
         return collectorItems;
     }
 
+    public List<CollectorItem> getCollectorItems(CollectorType type) {
+        return collectorItems.get(type);
+    }
+
     public void addCollectorItem(CollectorType collectorType, CollectorItem collectorItem) {
         // Currently only one collectorItem per collectorType is supported
-        collectorItems.put(collectorType, Arrays.asList(collectorItem));
+        if (collectorItems.get(collectorType) == null) {
+            collectorItems.put(collectorType, Arrays.asList(collectorItem));
+        } else {
+            List<CollectorItem> existing = collectorItems.get(collectorType);
+            List<CollectorItem> newList = new ArrayList<>(existing);
```

Comment:
Its actually not a temp... the `collectorItems.get()` returns a fixed length list. Need to do this to add more:  `List<CollectorItem> existing = new ArrayList<> (collectorItems.get(collectorType));` - instead I used a "temp".